### ISSUE TITLE
Remove Room Search from Global Search

### ISSUE THREAD

**Issue by rdorsch** [LiMium/mini-vector-android#28]
Hi,

many thanks for providing miniVector.

riot.im has an USK 12 age rating in Google PlayStore. This essentially keeps me from recommending riot instead of WA for children usage and parents from accepting riot.im for their childrens usage. mini-vector has no age restriction in PlayStore. From a content perspective this is not justified from my perspective. A global search in rooms with simple search terms reveals, that inappropriate content can be found easily.

I suggest to remove this global search in rooms in mini-vector to better justify the age rating.

Many thanks
Rainer

**Comment by Bubu** [comment]
This, I think, comes back to [banquets vs barbecue](https://blogs.gnome.org/tbernard/2018/05/16/banquets-and-barbecues/) again. MV, being a fork of riot is oriented more towards banquets. I find general room search valuable and I'd definitely like to have it in MV going forward (of course @hrj is the maintainer and free to disagree).

I think for a barbecue app (like pattle?) it makes a lot more sense to drop room search as most users won't need it.

Edit: wrong link...

**Comment by rdorsch** [comment]
Thanks for sharing the link. I agree child friendly is closer to barbecue than banquet, but it is not the same and deserves therefore a separate dimension.

E.g. if you search for [Pattle](https://play.google.com/store/apps/details?id=im.pattle.app&hl=en) you find an age rating of USK 18 years in the play store.

At least child friendliness is not covered by any client so far it seems, I certainly would appreciate if any matrix client would pickup this issue. And MV is at least a good candidate judging after its USK rating :-)

**Comment by hrj** [comment]
I have a few questions:

- Apart from global search, what other potential problems can be there in a matrix app from a child's perspective? Any system will be as bad as its weakest link, so it doesn't make sense to disable a random subset of features. Somebody needs to do a more comprehensive study and list the weakest links. A few examples below.
- Should invitations to join rooms from strangers be allowed?
- What about inappropriate content being shared by known contacts? How is that usually approached in child friendly software?
- The content and contact list is actually managed by the matrix server, and the client is just showing a reflection of the server data. Should the server also be made child friendly (whatever that means)?
- The idea of what is child friendly varies from region to region. Just FYI, Pattle's Play store listing in my country is shown as `12+`.

----

My point is, it's a big topic that needs a comprehensive plan before we disable features and call it a day.

I am open to hearing such a plan, but what seems most practical right now is Yet Another Fork ™
It will be a lot simpler to create a fork that disables all potentially harmful features, and let the user choose which app fork they want to use / promote.

**Comment by rdorsch** [comment]
I agree, listing the weakest link makes sense. Until yesterday, I did not know the room search, my suggestion is to have a candidate list e.g. in the wiki such that every expert can contribute ideas.

But when evaluating the list, at least for the first round I suggest to ignore everything which signal or WA implement. They are rated with USK 0 and I would be surprised if they didn't carefully investigate the topic.

**Comment by hrj** [comment]
> my suggestion is to have a candidate list e.g. in the wiki such that every expert can contribute ideas.

Sounds good. I have created a wiki page for this here: https://github.com/LiMium/mini-vector-android/wiki/Matrix-usability-for-young-people

Feel free to edit it yourself or invite more people to edit it. I have added some headings as a starting point. Feel free to change that too.
